crates/oxc_ast/src/ast/ts.rs

```diff
@@ -868,8 +868,8 @@ pub enum TSTypePredicateName<'a> {
     This(TSThisType),
 }
 
-#[visited_node]
-#[derive(Debug, Hash)]
+#[visited_node(scope(ScopeFlags::TsModuleBlock), enter_scope_before(body))]
```

Comment:
I'm wondering if `scope(ScopeFlags::TsModuleBlock), enter_scope_before(body)` is still necessary, as it seems that the `oxc_traverse` hasn't changed with it!

Reply:
`enter_scope_before(...)` is still used by the `oxc_traverse` codegen. You're right that the `scope(...)` part is ignored at present, but I thought best leave them there for now because it'll be needed for #3392.